Sources/protoc-gen-swift/OneofGenerator.swift

```diff
@@ -49,6 +50,53 @@ class OneofGenerator {
         self.swiftRelativeName = sanitizeOneofTypeName(descriptor.swiftRelativeType)
         self.swiftFullName = swiftMessageFullName + "." + swiftRelativeName
         self.comments = file.commentsFor(path: path)
+
+        let first = fieldsSortedByNumber.first!.number
+        let last = fieldsSortedByNumber.last!.number
+        // Easy case, all in order and no gaps:
+        if first + fields.count - 1 == last {
+            oneofIsContinuousInParent = true
+        } else {
+            // See if all the oneof fields were in order within the (even if there were number gaps).
+            //    message Good {
+            //      oneof o {
+            //        int32 a = 1;
+            //        int32 z = 26;
+            //      }
+            //    }
+            //    message Bad {
+            //      oneof o {
+            //        int32 a = 1;
+            //        int32 z = 26;
+            //      }
+            //      int32 m = 13;
+            //    }
+            let firstIndex = parentFieldNumbersSorted.index(of: first)!
+            var isContinuousInParent = true
+            for i in 0..<fields.count {
+                if fieldsSortedByNumber[i].number != parentFieldNumbersSorted[firstIndex + i] {
```

Comment:
The catch is the parent list includes the fields in this oneof, so it would be a little more complicated.

Another solution (which I didn't try coding) could be to see if sortedFields == parentFields[firstIndex..<firstIndex+fields.count].